translate/storage/markdown.py

```diff
@@ -0,0 +1,462 @@
+
+        # return translated text
+        return self.callback(text)
+
+
+class TranslatingMarkdownRenderer(MarkdownRenderer):
+    def __init__(self, translate_callback, *extras, max_line_length: int = None):
+        super().__init__(*extras, max_line_length=max_line_length)
+        self.translate_callback = translate_callback
+        self.bypass = False
+        self.path = []
+
+    def render(self, token: mistletoe.token.Token) -> str:
+        try:
+            # set the root node here, because the rendering also involves some parsing,
+            # and the parsing requires a valid root node.
+            mistletoe.token._root_node = token
+            return super().render(token)
+        finally:
+            mistletoe.token._root_node = None
+
+    # rendering of span tokens:
+    # override to inject placeholders and translate content
+
+    _leading_ws = re.compile(r"^(\s+)\S")
+    _trailing_ws = re.compile(r"\S(\s+)$")
+
+    def render_raw_text(self, token: span_token.RawText) -> Iterable[Fragment]:
+        # make separate fragments for leading and trailing space, if applicable
+        match = self._leading_ws.search(token.content)
+        if match:
+            yield Fragment(match.group(1), wordwrap=True)
+            token.content = token.content.lstrip()
+        match = self._trailing_ws.search(token.content)
+        if match:
+            yield Fragment(token.content.rstrip(), wordwrap=True)
+            yield Fragment(match.group(1), wordwrap=True)
+        else:
+            yield Fragment(token.content, wordwrap=True)
+
+    def render_auto_link(self, token: span_token.AutoLink) -> Iterable[Fragment]:
+        # replace with placeholder, unless in bypass mode
+        if self.bypass:
+            yield from super().render_auto_link(token)
+            return
+
+        yield Fragment(None, placeholder_content=super().render_auto_link(token))
+
+    def render_line_break(self, token: span_token.LineBreak) -> Iterable[Fragment]:
+        if self.bypass:
+            yield from super().render_line_break(token)
+            return
+
+        yield Fragment("\n", wordwrap=token.soft, hard_line_break=not token.soft)
+
+    def render_html_span(self, token: span_token.HTMLSpan) -> Iterable[Fragment]:
+        # replace with placeholder, unless in bypass mode
+        if self.bypass:
+            yield from super().render_html_span(token)
+            return
+
+        yield Fragment(None, placeholder_content=super().render_html_span(token))
+
+    def render_link_or_image(
+        self, token: span_token.SpanToken, target: str
+    ) -> Iterable[Fragment]:
+        if self.bypass:
+            yield from super().render_link_or_image(token, target)
+            return
+
+        yield from self.embed_span(Fragment("["), token.children, Fragment("]"))
+
+        if token.dest_type in ["uri", "angle_uri"]:
+            # "[" description "](" dest_part [" " title] ")"
```

Comment:
The comments in this method are not really explaining. Either remove them or clarify what they describe.